### PR TITLE
handle 403 with an empty body

### PR DESCRIPTION
**Pull request by dmccraw**

### CODE REVIEW

**Comment by glanotte** [comment]
👍 on green

**Comment by semmons99** [comment]
Do we need any version bump and/or release to rubygems? I can add you as an owner if you need access.

**Comment by dmccraw** [comment]
I was able to release it to ruby_gems. It went out as 2.0.0beta6.

File: lib/teamsnap/api.rb [hunk 1]
```diff
@@ -73,6 +73,8 @@ def self.via(klass, method)
 
     def self.parse_error(resp)
       return "Object Not Found (404)" if resp.status == 404
+      return "Forbidden (403)" if resp.status == 403 && resp.body == ""
```
Comment (coreycondardo) [inline lib/teamsnap/api.rb]: Is there a reason why we are not returning `Forbidden (403)` when `resp.status == 403` without an empty body?

Comment (dmccraw) [inline lib/teamsnap/api.rb]: I was just following the 404 behavior. I don't have a preference either way.

Comment (coreycondardo) [inline lib/teamsnap/api.rb]: 404 doesn't check for `resp.body == ""`. 
What I mean is why don't we have this:
```
return "Forbidden (403)" if resp.status == 403 || resp.status == 403 && resp.body == ""
```
That way on any 403 it returns the proper string, not just a body that is empty. Or will the body _always_ be empty on a 403?

Comment (dmccraw) [inline lib/teamsnap/api.rb]: We have some cases in apiv3 where a 403 does return it's own body so in those cases I want to return what apiv3 generated. I only need to check if there isn't a body.
I wonder if 404 should be checking for an empty body. There are some instances of apiv3 returning a 404 and having an error message. @glanotte

**Pull request merged**